kivy/network/newurlrequest.py

```diff
@@ -0,0 +1,584 @@
+'''
+UrlRequest
+==========
+
+.. versionadded:: 1.0.8
```

Comment:
need to update the name and version added.

But more generally, i think it would be better to avoid so much duplication, and rather to have one class inheriting from the other, (and possibly an abstract implementation), as it needs a better name than `newurlrequest` i think something like `urlrequest_requests.py` would be better, even i the name of the class stays the same so only import needs to be changed for it to be a drop-in replacement.

Reply:
I agree it would be better to only have one module, hence i think having in the same module, the base implementation, and the two subclasses, but making `UrlRequest` an alias to `UrllibUrlRequest` would be a good option, people can still import either class if they want to, and we can change the alias to point to `RequestsUrlRequest` in 1-2 releases, deprecating the legacy version, (but still available ,if people did something weird and have compatibility issues), and then removing it entirely further down the line, so people have enough headway to do any updates they need.